api-v2/src/student/student.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+  DeleteResult,
+  FindOneOptions,
+  Repository,
+  UpdateResult,
+} from 'typeorm';
+import { CreateStudentDto } from './dto/create-student.dto';
+import { UpdateStudentDto } from './dto/update-student.dto';
+import { Student } from './entities/student.entity';
+import { StudentNotFoundError } from './student-not-found.error';
+
+@Injectable()
+export class StudentService {
+  constructor(
+    @InjectRepository(Student)
+    private studentRepository: Repository<Student>,
+  ) {}
+
+  async create(createStudentDto: CreateStudentDto): Promise<Student> {
+    // make sure the user doesn't already exists
+    const { email } = createStudentDto;
+    const userInDb = await this.studentRepository.findOne({ where: { email } });
+    if (userInDb) {
+      throw new Error('A user with the email is already registered');
+    }
+
+    const newStudent = this.studentRepository.create(createStudentDto);
+    try {
+      return this.studentRepository.save(newStudent);
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  findAll(): Promise<Student[]> {
```

Comment:
im too lazy to lookup where this is used, but when would we ever need this? it should probably be paginated (there will potentially be thousands of students)

Reply:
not used anywhere! just used for dev routes that I used for debugging. I'll just remove the dev routes + any service methods associated to them to avoid confusion

Reply:
Nah i think it's fine to keep dev routes, but deleting unised service methods sounds good to me

Reply:
ah cool. well we'll have to keep this one since it's used by a dev route... i'll do a pass and make sure we don't have unused service methods too!